cniconf/watcher.go

```diff
@@ -125,3 +127,29 @@ func (w *watcher) apply(network metadata.Network) error {
 
 	return lastErr
 }
+
+func UpdateCNIConfigByHost(config interface{}, host metadata.Host) interface{} {
+	props, isMap := config.(map[string]interface{})
+	if !isMap {
+		return config
+	}
+
+	for aKey, aValue := range props {
+		v, isString := aValue.(string)
+		if !isString {
+			props[aKey] = UpdateCNIConfigByHost(aValue, host)
+		} else {
+			if strings.Contains(v, "__host_label__") {
+				label := strings.TrimSpace(strings.Split(v, ":")[1])
```

Comment:
Please check the length before accessing the array by index. This can potentially cause a crash if there is a problem with config.